### DISCUSSION TITLE
tf-idf를 적용할 때 길이가 1133이어도 괜찮은지?

### DISCUSSION THREAD

**Discussion by Ssunbell** [BoocampMonster/Competition_MRC#3] [General]
<img width="757" alt="스크린샷 2022-12-26 오후 4 30 01" src="https://user-images.githubusercontent.com/97590480/209518492-044a21cc-04dd-49b5-997c-4408e4d72177.png">

<img width="995" alt="스크린샷 2022-12-26 오후 3 46 25" src="https://user-images.githubusercontent.com/97590480/209513866-29338911-852e-42d6-852b-dd48c2c52ddf.png">

sparse retrieval을 진행할 때 tf-idf vector를 생성하는 동안 길이가 1133이라서 짤릴 수 있다고 경고를 날림. 이것이 문제가 되지는 않을까??

**Comment by chanmuzi** [comment]
## Before Tokenizing

[ wiki 데이터에서 추출한 context의 길이 정보 ]
<img width="500" alt="image" src="https://user-images.githubusercontent.com/101971295/209558015-cab35ea5-7c6b-4c1b-8608-c8f6ed5ffd77.png"> 

[ train / valid 데이터에서 추출한 context의 길이 정보 ]
<img width="500" alt="image" src="https://user-images.githubusercontent.com/101971295/209558340-21a8ea73-d2a6-46d2-b03f-ada1dd7538e6.png">


## After Tokenizing
[ wiki 데이터에서 추출한 context의 길이 정보 ]
<img width="500" alt="image" src="https://user-images.githubusercontent.com/101971295/209567275-6af9f6c3-d99d-44ca-a1e8-85f791ca5485.png"> 


[ train / valid 데이터에서 추출한 context의 길이 정보 ]
<img width="500" alt="image" src="https://user-images.githubusercontent.com/101971295/209567250-dc553bf4-df0d-47bc-825b-05e94626e62b.png"> 


확인 결과 빨간 박스로 표시한 부분에서 warning이 발생하는 것이 맞더라구.
위의 두 개 사진은 tokenize 적용 전인데 길이가 꽤 길어서 문제가 되지 않을까 싶었는데, tokenize 이후엔 생각보다 길이가 많이 줄어들어서 max_len을 초과하지 않게 되는 case가 대부분이야.
지금 상황에서는 너가 지적한 문제가 발생하고 있을 가능성이 매우 높다고 생각이 들어.

이 상황에서 눈에 뜨인 건 세 가지인데,
1. outlier의 존재가 wiki 데이터에서 유난히 두드러진다.
2. max_len을 초과하지 않는 안정적인 분포는 오히려 wiki 데이터에서 나타난다.(표준편차가 크긴 하지만 outlier 제거하면 많이 줄어들 것으로 예상)
3. 75% 기준치가 max_len을 한참 초과하는 train/vaild 데이터는 제대로 추론하지 못할 가능성이 높다.

따라서 max_len을 초과하는 context들은 어떤 것들이고 개수는 어떤지 직접 확인할 필요가 있을 것 같아.
얘기했던 것처럼 길이가 가장 긴 50개의 context도 wiki 기준으로 한 번 뽑아 보도록 할게!
